woocommerce-services.php

```diff
@@ -645,13 +645,40 @@ public function attach_hooks() {
 			add_action( 'admin_init', array( $this, 'load_admin_dependencies' ) );
 			add_filter( 'wc_connect_shipping_service_settings', array( $this, 'shipping_service_settings' ), 10, 3 );
 			add_action( 'woocommerce_email_after_order_table', array( $this, 'add_tracking_info_to_emails' ), 10, 3 );
+			add_action( 'save_post', array( $this, 'maybe_trigger_product_notice' ), 10, 2 );
 
 			$tracks = $this->get_tracks();
 			$tracks->init();
 
 			$this->taxjar->init();
 		}
 
+		public function maybe_trigger_product_notice( $post_id, $post ) {
+			if (
+				'publish' !== $post->post_status ||
+				'product' !== $post->post_type ||
```

Comment:
You can avoid this check by hooking into `save_product` instead of `save_post`. See: https://github.com/WordPress/WordPress/blob/master/wp-includes/post.php#L3724